packages/code-du-travail-frontend/pages/dossiers/[slug].js

```diff
@@ -46,50 +47,44 @@ function DossierThematique({ dossier }) {
       <Section>
         <Container narrow>
           <PageTitle subtitle={description}>{title}</PageTitle>
-          <SelectWrapper>
-            <Select
-              onChange={(event) => {
-                setFilter(event.target.value);
-              }}
-            >
-              <option value="">Tous les contenus</option>
-              {sortedCategories.map((category) => (
-                <option key={category.id} value={category.id}>
-                  {category.shortTitle || category.title}
-                </option>
-              ))}
-            </Select>
-          </SelectWrapper>
         </Container>
         <MainContainer>
           <FixedWrapper>
             <NavTitle>Sommaire</NavTitle>
             <TableOfContent
-              ids={sortedCategories.map((category) => category.id)}
+              contents={sections.flatMap(({ label, categories }) => [
```

Comment:
can you explain the difference with map here, not sure to understand

Reply:
`TableOfContent` takes a flat array, without the flat map we would return an arrray of arrays each containg several objects that could have a label key, an id key, or none.

Reply:
```
type Section = {
  label?: string,
  categories: {
    id: string
  }[],
};

type Contents = {label?: string, id?:string};

const sections : Section[] = [{label:"youhou", categories: [{id: "gdgfzga"}]}];

const contents : Contents[] = sections.flatMap(({ label, categories }) => [
  { label },
  ...categories.map((category) => ({
    id: category.id,
  })),
]);
```
You can check with this in the TS Playground by replace flatMap with map

Reply:
thanks it's perfect !